sway/input/text_input.c

```diff
@@ -280,10 +539,13 @@ static void relay_handle_input_method(struct wl_listener *listener,
 	wl_signal_add(&relay->input_method->events.destroy,
 		&relay->input_method_destroy);
 	relay->input_method_destroy.notify = handle_im_destroy;
+	wl_signal_add(&relay->input_method->events.new_popup_surface,
+		&relay->input_method_new_popup_surface);
+	relay->input_method_new_popup_surface.notify = handle_im_new_popup_surface;
```

Comment:
This is missing logic to remove the listener when the IM is destroyed.

I noticed that we're missing the same thing for the `input_method_destroy` as well: https://github.com/swaywm/sway/pull/7988

Reply:
Ok, am I doing right?

Reply:
Yup!